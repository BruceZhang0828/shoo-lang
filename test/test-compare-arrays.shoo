array<int> a = [1,2,3,4,5];
array<int> b = [1,2,3,4,5];
array<int> c = [5,4,3,2,1];
array<int> d = [2,3];

function temp(array<int> g) void{
    if (g == a) {
        println("same ptr");
    } else {
        println("diff ptr");
    }
}

temp(a);

if (a == a) {
    println("same");
} else {
    println("what");
}

if (a == b) {
    println("same values");
} else {
    println("same value not same ptr"); 
}

if (a == c) {
    println("equal?");
} else {
    println("not equal at all");
}

if (b == d) {
    println("no");
} else {
    println("yes");
}



