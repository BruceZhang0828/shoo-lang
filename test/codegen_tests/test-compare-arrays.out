same ptr
same
same value not same ptr
not equal at all
yes
